app/controllers/v1/supplemental_claims_controller.rb

```diff
@@ -87,28 +91,29 @@ def handle_personal_info_error(e)
 
     def process_submission
       req_body_obj = request_body_hash.is_a?(String) ? JSON.parse(request_body_hash) : request_body_hash
+      saved_claim_request_body = req_body_obj.to_json # serialize before request body is modified
       form4142 = req_body_obj.delete('form4142')
       sc_evidence = req_body_obj.delete('additionalDocuments')
       zip_from_frontend = req_body_obj.dig('data', 'attributes', 'veteran', 'address', 'zipCode5')
+
       sc_response = decision_review_service.create_supplemental_claim(request_body: req_body_obj, user: @current_user)
       submitted_appeal_uuid = sc_response.body.dig('data', 'id')
-      unless submitted_appeal_uuid.nil?
-        ActiveRecord::Base.transaction do
-          appeal_submission_id = create_appeal_submission(submitted_appeal_uuid, zip_from_frontend)
-
-          ::Rails.logger.info(post_create_log_msg(appeal_submission_id:, submitted_appeal_uuid:))
-          if form4142.present?
```

Comment:
ok, so this check isn't necessary anymore?

Reply:
I moved the check into the `handle_4142` method and made it into a guard clause, line 45